1.regression_modelling/scripts/3.find_sc_crops.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Generate single-cell crops for min/max montages of the top feature (Zernike phase 2,2) for both stains
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+from pprint import pprint
+from typing import List
+
+import cv2
+import pandas as pd
+
+
+# ## Set functions for processing and generating single-cell crops
+
+# In[2]:
+
+
+# Function for formatting min/max row data frames into dictionaries
+def create_sc_dict(dfs: List[pd.DataFrame], names: List[str]) -> dict:
+    """Format lists of data frames and names into a dictionary with all relevant metadata to find single-cell images.
+
+    Args:
+        dfs (List[pd.DataFrame]): List of data frames each containing a single cell and relevant metadata.
+        names (List[str]): List of names corresponding to the data frames.
+
+    Returns:
+        dict: Dictionary containing info relevant for finding single-cell crops.
+    """
+    sc_dict = {}
+    for df, name in zip(dfs, names):
+        for i, (_, row) in enumerate(df.iterrows()):
+            if i >= 5:  # Limit to 5 rows
+                break
+            key = f"{name}_{i+1}"  # Create key with incrementing number
+            sc_dict[key] = {
+                "plate": row["Metadata_Plate"],
+                "well": row["Metadata_Well"],
+                "site": row["Metadata_Site"],
+                "location_center_x": row["Metadata_Nuclei_Location_Center_X"],
+                "location_center_y": row["Metadata_Nuclei_Location_Center_Y"],
+            }
+    return sc_dict
+
+
+# In[3]:
+
+
+# Function for generating and saving single-cell crops per channel as PNGs
+def generate_sc_crops(
```

Comment:
Seeing how these functions build off eachother. Consider maybe making these class methods in a class. No pressure if not.

Reply:
I think for now I will keep it as two functions for now. I will consider class in the future if I plan to build more into generating these crops. I appreciate the suggestion!